docs/integrations/scope.mdx

```diff
@@ -0,0 +1,157 @@
+
+```tsx
+import {
+  Provider as ScopedProvider,
+  useAtom as useScopedAtom,
+} from 'jotai-scope'
+import { atom, useAtom } from 'jotai'
+import {
+  Provider as ScopedProvider,
+  useAtom as useScopedAtom,
+} from 'jotai-scope'
+
+const countAtom = atom(0)
+const anotherCountAtom = atom(0)
+
+const Counter = () => {
+  const [count, setCount] = useAtom(countAtom)
+  const [anotherCount, setAnotherCount] = useScopedAtom(anotherCountAtom)
+  return (
+    <>
+      <div>
+        <span>count: {count}</span>
+        <button type="button" onClick={() => setCount((c) => c + 1)}>
+          increment
+        </button>
+      </div>
+      <div>
+        <span>another count: {anotherCount}</span>
+        <button type="button" onClick={() => setAnotherCount((c) => c + 1)}>
+          increment
+        </button>
+      </div>
+    </>
+  )
+}
+
+const App = () => (
+  <div>
+    <h1>First Provider</h1>
+    <ScopedProvider atoms={[anotherCountAtom]}>
+      <Counter />
+    </ScopedProvider>
+    <h1>Second Provider</h1>
+    <ScopedProvider atoms={[anotherCountAtom]}>
+      <Counter />
+    </ScopedProvider>
+  </div>
+)
+```
+
+<CodeSandbox id="k5gfvl" />
+
+## `createIsolation`
+
+Both Jotai's Provider and `jotai-scope`'s scoped provider
+are still using global contexts.
+
+If you are developing a library that depends on Jotai and
+the library user may use Jotai separately in their apps,
+they can share the same context. This can be troublesome
+because they point to unexpected Jotai stores.
+
+To avoid conflicting the contexts,
+a utility function called `createIsolation` is exported from `jotai-scope`.
+
+Check out the example in the repo: https://github.com/jotaijs/jotai-scope/tree/main/examples/01_isolation
+
+## `bunshi` (formerly `jotai-molecules`)
+
+Jotai atoms provide a basic solution to optimize re-renders.
+Atoms defined globally can depend on other atoms,
+but they can't depend on props and state within a component tree.
+It's possible to define atoms within a component tree,
+but then you would need to pass those atoms in some ways
+(for example, [atoms-in-atom](../guides/atoms-in-atom.mdx).)
+
+[bunshi](https://github.com/saasquatch/bunshi) is
+a third-party library to help such use cases.
+
+See [Motivation](https://github.com/saasquatch/bunshi/tree/v1.1.1#motivation) for more details.
+
+### Install
+
+```
+yarn add bunshi
+```
+
+### Counter Example
+
+```jsx
+import { atom, useAtom } from 'jotai'
+import { molecule, useMolecule, createScope, ScopeProvider } from 'bunshi/react'
+
+const InitialCountScope = createScope(0)
+const countMolecule = molecule((getMol, getScope) => {
+  const initialCont = getScope(InitialCountScope)
+  return atom(initialCont)
+})
+
+const Counter = () => {
+  const countAtom = useMolecule(countMolecule)
+  const [count, setCount] = useAtom(countAtom)
+  return (
+    <div>
+      {count} <button onClick={() => setCount((c) => c + 1)}>+1</button>
+    </div>
+  )
+}
+
+const App = () => (
```

Comment:
I'm currently working on improving bunshi docs and I think that maybe this [example](https://github.com/saasquatch/bunshi/issues/26#issuecomment-1751805315) helps convey how bunshi works a little better. I don't think that many users will want counters based on initial value to share state.

Reply:
Hi @DavidJFelix,

I'd admit the docs and examples have huge room for improvement. Hope you can help us. I'll merge this PR. Please feel free to open a new PR to improve the example.